### PR TITLE
Feature/docker.volume

### LINKED ISSUE #299: docker.volume resource
Ability to manage docker data volumes: https://docs.docker.com/engine/reference/commandline/volume_create/

### PR DESCRIPTION
**Pull request by ryane**
fixes #299

### CODE REVIEW

**Comment by stevendborrelli** [comment]
Getting a panic when I try to run the sample:
```
./converge plan --local samples/dockerVolume.hcl
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x421382]

goroutine 22 [running]:
panic(0x6308e0, 0xc4200120a0)
    /usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1a1
github.com/asteris-llc/converge/resource/docker/volume.(*Volume).Check(0xc420175080, 0xa10100, 0xc42015d900, 0x20, 0xc42015d900, 0x0, 0x0)
    /Users/steve/go/src/github.com/asteris-llc/converge/resource/docker/volume/volume.go:65 +0x3c2
```

**Comment by ryane** [comment]
panic should be fixed + rebased

File: resource/docker/volume/volume.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+		v.Status.Level = resource.StatusFatal
+		return v, err
+	}
+
+	v.Status.AddDifference(v.Name, string(volumeState(vol)), string(v.State), "")
+
+	if v.State == StatePresent && vol != nil && v.Force {
+		expectedLabels := mapToString(v.Labels)
+		actualLabels := mapToString(vol.Labels)
+		v.Status.AddDifference("labels", actualLabels, expectedLabels, "")
```
Comment (stevendborrelli) [inline resource/docker/volume/volume.go]: Shouldn't we only add these diffs if they don't match?

Comment (ryane) [inline resource/docker/volume/volume.go]: it's an option but this approach is consistent with the existing docker resources. this is discussed more in #325. let me know if you think I should change it.

we could also update the human printer to only print diffs with changes. something like:

```diff
diff --git a/prettyprinters/human/human.go b/prettyprinters/human/human.go
index 74d41ac..4e137a8 100644
--- a/prettyprinters/human/human.go
+++ b/prettyprinters/human/human.go
@@ -136,7 +136,9 @@ func (p *Printer) DrawNode(g *graph.Graph, id string) (pp.Renderable, error) {
 	Has Changes: {{if .HasChanges}}{{yellow "yes"}}{{else}}no{{end}}
 	Changes:
 		{{- range $key, $values := .Changes}}
+		{{- if $values.Changes}}
 		{{cyan $key}}:	{{diff ($values.Original) ($values.Current)}}
+		{{- end}}
 		{{- else}} No changes {{- end}}
 
 `)
```

**Pull request merged**